hearpreprocess/util/audio.py

```diff
@@ -3,129 +3,178 @@
+    All the above are only done when the input file doesnot match the required
+    conditions of duration, streams and extension.
+    In case, we are unable to get any audio stats, by default all the steps are done.
+
+    If the audio is already satisfies all the expected filters, the step is skipped
+    and a symlink is created to the original file
     """
-    ret = subprocess.call(
-        [
-            "ffmpeg",
-            "-y",
-            "-i",
-            str(in_file),
-            "-filter_complex",
-            f"apad=whole_dur={duration},atrim=end={duration}",
-            "-ac",
-            "1",
-            "-c:a",
-            "pcm_f32le",
-            str(out_file),
-        ],
-        stdout=subprocess.DEVNULL,
-        stderr=subprocess.DEVNULL,
-    )
-    # Make sure the return code is 0 and the command was successful.
-    assert ret == 0, f"ret = {ret}"
-
-
-def convert_to_mono_wav(in_file: str, out_file: str):
-    devnull = open(os.devnull, "w")
-    # If we knew the sample rate, we could also pad/trim the audio file now, e.g.:
-    # ffmpeg -i test.webm -filter_complex \
-    #    apad=whole_len=44100,atrim=end_sample=44100 \
-    #    -ac 1 -c:a pcm_f32le ./test.wav
-    # print(" ".join(["ffmpeg", "-y", "-i", in_file,
-    #    "-ac", "1", "-c:a", "pcm_f32le", out_file]))
-    ret = subprocess.call(
-        ["ffmpeg", "-y", "-i", in_file, "-ac", "1", "-c:a", "pcm_f32le", out_file],
-        stdout=devnull,
-        stderr=devnull,
-    )
-    # Make sure the return code is 0 and the command was successful.
-    assert ret == 0
+    # Get the audio stats for the audio file
+    audio_stats = get_audio_stats(in_file)
+    # Get the filters to apply to the audio. 
+    # If ffmpeg probe is unable to fetch the audio stats
+    # in which case the stats will be empty
+    # we will make all the filters as True
+    if audio_stats:
+        duration_filter_flag = audio_stats["duration"] != duration
+        mono_filter_flag = not audio_stats["mono"]
+    else:
+        duration_filter_flag = mono_filter_flag = True
+
+    ext_filter_flag = Path(in_file).suffix.lower() != ".wav"
+
+    # If the audio has the desired duration and is already mono .wav all the flags will
+    # be false, then we will move to the else part where we will just create a symlink
+    if duration_filter_flag or mono_filter_flag or ext_filter_flag:
+        # create a ffmpeg command chain to take the input
+        cmd_chain = ffmpeg.input(in_file).audio
+        # Add duration commands if required
+        if duration_filter_flag:
+            cmd_chain = cmd_chain.filter("apad", whole_dur=duration).filter(
+                "atrim", end=duration
+            )
+        # Add mono flag if required
+        if mono_filter_flag:
+            cmd_chain = cmd_chain.output(out_file, f="wav", acodec="pcm_f32le", ac=1)
+        else:
+            cmd_chain = cmd_chain.output(out_file, f="wav", acodec="pcm_f32le")
```

Comment:
If you have to specify the output then just specify it is mono regardless and remove the if